virttest/qemu_vm.py

```diff
@@ -1038,10 +1038,35 @@ def add_memorys(devices, params):
                 options.append("maxmem=%s" % mem_params["maxmem"])
                 if mem_params.get("slots"):
                     options.append("slots=%s" % mem_params["slots"])
+            sgx_epc_memdev_id = dict()
             for name in params.objects("mem_devs"):
                 dev = devices.memory_define_by_params(params, name)
+                for ele in dev:
+                    if isinstance(ele, qdevices.Memory) and \
+                            ele.params["backend"] == "memory-backend-epc":
+                        sgx_epc_memdev_id[name] = ele.params["id"]
                 devs.extend(dev)
+            sgx_epc_cmd = ""
+            if params.get("vm_sgx_epc_devs"):
+                cap_type = "sgx-epc"
+                for idx, ele in enumerate(params.objects(
+                        "vm_sgx_epc_devs")):
+                    sgx_epc_params = params.object_params(ele)
+                    opt_mapping = {
+                        'memdev': sgx_epc_memdev_id[
+                            sgx_epc_params.get("vm_sgx_epc_memdev")],
+                        'node': sgx_epc_params.get("vm_sgx_epc_node")
```

Comment:
use sgx_pec_params["vm_sgx_epc_memdev"] and sgx_epc_params["vm_sgx_epc_node"] instead, for you mentioned in base.cfg that these two params are required, so if any of them is not set, VT will not run(skipped? I forgot it, but a msg will show).

And then at line1061, we can remove if val